opentimelineio_contrib/adapters/advanced_authoring_format.py

```diff
@@ -862,6 +861,34 @@ def _contains_something_valuable(thing):
     return True
 
 
+def attach_markers(timeline):
+    '''
+    Search for markers and attach them to their corresponding track
+    '''
+    markers_dict = {}
+
+    # Get all markers in the given unsimplified timeline. Markers come in as non
+    # audio or video tracks
+    for track in timeline.each_child(descended_from_type=otio.schema.Track):
+        if track.markers:
+            for m in track.markers:
+                # Using pop() will remove the value from the set causing
+                # incomplete metadata for DescribedSlots
+                d_slot = next(iter(m.metadata.get('AAF').get('DescribedSlots')))
```

Comment:
if `metadata.get('AAF').get('DescribedSlots')` is `None` this will raise an exception.  Does this always get populated by the adapter?

Reply:
Yeah every marker should have DescribedSlots in the metadata. This indicates which slot the marker corresponds to. If that isn't there, then I think raising an exception is appropriate. Instead of the NoneType is not iterable exception, I could check for this case and raise another exception with a more meaningful message?

Reply:
That sounds good.